flaml/automl/automl.py

```diff
@@ -3557,6 +3569,25 @@ def _search_sequential(self):
                     state.best_config,
                     self.data_size_full,
                 )
+                if getattr(self._trained_estimator, "params", {}) and getattr(
+                    self._trained_estimator, "ITER_HP", None
+                ):
+                    _hp_trained_iter = self._trained_estimator.params.get(
+                        self._trained_estimator.ITER_HP
+                    )
+                    _best_config_iter = self.best_config.get(
+                        self._trained_estimator.ITER_HP
+                    )
+                    if _hp_trained_iter != _best_config_iter:
+                        logger.warning(
+                            "Early stopping happened when retraining a model with the best configuration."
+                            f" The best config's {self._trained_estimator.ITER_HP} is {_best_config_iter}"
+                            f" and the actual {self._trained_estimator.ITER_HP} used for retraining the model is {_hp_trained_iter}."
+                            " This early stopping happens because flaml needs to do its best effort to"
+                            " retrain without violating the time budget when retrain_full is set to 'budget'. "
+                            " If this mismatch is not desired, please set retrain_full to True."
+                        )
```

Comment:
Please confirm the code coverage for this part.

Reply:
<img width="517" alt="Screenshot 2023-01-05 at 6 49 04 AM" src="https://user-images.githubusercontent.com/62365520/210773934-5d93ebe9-416c-406d-b1fa-e7b827b2c052.png">